HopsanGUI/HcomHandler.cpp

```diff
@@ -363,7 +363,10 @@ HcomHandler::HcomHandler(TerminalConsole *pConsole) : QObject(pConsole)
     registerInternalFunction("lp1", "Applies low-pass filter of first degree to vector","Usage: lp1(vector, frequency)\nUsage: lp1(vector, timevector, frequency)");
     registerInternalFunction("ddt", "Differentiates vector with respect to time (or to custom vector)","Usage: ddt(vector)\nUsage: ddt(vector, timevector)");
     registerInternalFunction("int", "Integrates vector with respect to time (or to custom vector)", "Usage: int(vector)\nUsage: int(vector, timevector)");
-    registerInternalFunction("fft", "Generates frequency spectrum plot from vector","Usage: fft(vector)\nUsage: fft(vector, power[true/false])\nUsage: fft(vector, timevector)\nUsage: fft(vector, timevector, power[true/false])");
+    registerInternalFunction("esd", "Generates energy spectral density from vector","Usage: esd(vector, timevector [optional], windowing[rectangular/flattop/hann, optional], mintime[optional], maxtime[optional])\n");
```

Comment:
For backwards compatibility maybe keep the original fft function, but add a deprecation warning message when calling it.
And make it call one of the new functions

Reply:
Ok, I rewrote the fft functions it calls one of the other three, depending on the arguments.